### PR TITLE
Adding io metrics collection, environment values into metric name and general docker info stats

### PR DESCRIPTION
**Pull request by alcasim**
## Pull Request Checklist

**Is this in reference to an existing issue?**

#### General

- [x] Update Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/)

- [ ] Update README with any necessary configuration snippets

- [ ] Binstubs are created if needed

- [x] RuboCop passes

- [ ] Existing tests pass

#### New Plugins

- [ ] Tests

- [ ] Add the plugin to the README

- [ ] Does it have a complete header as outlined [here](http://sensu-plugins.io/docs/developer_guidelines.html#coding-style)

#### Purpose
Adding metrics-docker-info.rb with general information related to Docker (total number of containers, images, go routines, file descriptors...). These metrics mimics how telegraf gathers information from the docker plugin, and can be useful for people evaluating Sensu as source for their metrics.

In metrics-docker-stats.rb I have added two features:
io metric collection from docker virtual devices, in telegraf style also
Including some environment variables values into the metric (another feature from telegraf).

Please find some output for these changes in https://gist.github.com/alcasim/d21be7ff09d519d5aadb9d5d8215ec86

Also those changes performed on existing metric check (metrics-docker-stats.rb) are optional, so users won't be affected by including new flags into the checks

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,10 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 This CHANGELOG follows the format listed at [Keep A Changelog](http://keepachangelog.com/)
 
+### Added
```
Comment (majormoses) [inline CHANGELOG.md]: Can you move this under `## [Unreleased]`

File: bin/metrics-docker-info.rb [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+         description: 'Docker socket to connect. TCP: "host:port" or Unix: "/path/to/docker.sock" (default: "127.0.0.1:2375")',
+         short: '-H DOCKER_HOST',
+         long: '--docker-host DOCKER_HOST',
+         default: '127.0.0.1:2375'
+
+  option :docker_protocol,
+         description: 'http or unix',
+         short: '-p PROTOCOL',
+         long: '--protocol PROTOCOL',
+         default: 'http'
```
Comment (majormoses) [inline bin/metrics-docker-info.rb]: Unix sockets are the default and enabling the http interface takes some work, I think we should use a unix socket by default. But I am open to counter arguments.

Comment (alcasim) [inline bin/metrics-docker-info.rb]: Not a problem, we're using unix socket as well, I just copied same argument than for other metrics scripts (which uses http as default), I prefer to have unix socket as default IMHO.

File: bin/metrics-docker-info.rb [hunk 3]
```diff
@@ -0,0 +1,92 @@
+#! /usr/bin/env ruby
+#
+#   metrics-docker-stats
```
Comment (majormoses) [inline bin/metrics-docker-info.rb]: metrics-docker-info

File: bin/metrics-docker-info.rb [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+#
+# DESCRIPTION:
+#
+# Supports the stats feature of the docker remote api ( docker server 1.5 and newer )
```
Comment (majormoses) [inline bin/metrics-docker-info.rb]: info

File: bin/metrics-docker-info.rb [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+#   Gather stats from all containers on a host using socket:
+#   metrics-docker-stats.rb -p unix -H /var/run/docker.sock
+#
+#   Gather stats from all containers on a host using TCP:
+#   metrics-docker-stats.rb -p http -H localhost:2375
+#
+#   Gather stats from a specific container using socket:
+#   metrics-docker-stats.rb -p unix -H /var/run/docker.sock -c 5bf1b82382eb
+#
+#   See metrics-docker-stats.rb --help for full usage flags
```
Comment (majormoses) [inline bin/metrics-docker-info.rb]: metrics-docker-info

Comment (alcasim) [inline bin/metrics-docker-info.rb]: I apologize for that, pushing changes in some minutes.

Comment (majormoses) [inline bin/metrics-docker-info.rb]: np, copy/paste is super helpful.

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-docker/versions/1.4.0

**Pull request merged**